src/bin/cargo/commands/build.rs

```diff
@@ -50,14 +50,45 @@ pub fn exec(gctx: &mut GlobalContext, args: &ArgMatches) -> CliResult {
     let mut compile_opts =
         args.compile_options(gctx, CompileMode::Build, Some(&ws), ProfileChecking::Custom)?;
 
-    if let Some(out_dir) = args.value_of_path("out-dir", gctx) {
-        compile_opts.build_config.export_dir = Some(out_dir);
-    } else if let Some(out_dir) = gctx.build_config()?.out_dir.as_ref() {
-        let out_dir = out_dir.resolve_path(gctx);
-        compile_opts.build_config.export_dir = Some(out_dir);
+    // If the user specifies `--artifact-dir`, use that
+    let mut artifact_dir = args.value_of_path("artifact-dir", gctx);
+
+    // `--out-dir` is deprecated, but still supported for now
+    if artifact_dir.is_none() {
```

Comment:
Shouldn't the `if/else-if` approach still work?  Keeping the structure would make it easier to read and this embeds higher level context into the code structure (makes it clear to readers that they are mutually exclusive)

Reply:
You could even combine the approaches and do
```rust
compile_opts.build_config.export_dir = if let Some(artifact_dir) = ...
};
```

Reply:
I've changed it back to the old code style.

The if-expression style may not be equivalent to the existing code--right now `compile_opts.build_config.export_dir` is preserved if none of the command-line flags or config flags are present, whereas the if-expression would unconditionally reassign (in the final `else`, we *could* just assign it to its existing value, but that feels too "clever"). I don't *think* `compile_opts.build_config.export_dir` is set by any other code, but I'd rather not risk it.